### PR TITLE
add description field to accessKey

### PR DESCRIPTION
**Pull request by OfekAvergil**
## Related Issues

Related https://github.com/descope/etc/issues/6938

## Description

Adding the description field to accessKey (create and update).
## Must

- [X] Tests
- [X] Documentation (if applicable)

### CODE REVIEW

File: descope/management/access_key.py [hunk 1]
```diff
@@ -128,13 +132,14 @@ def update(
         Args:
         id (str): The id of the access key to update.
         name (str): The updated access key name.
+        description(str): an optional text the access key can hold.
```
Comment (nmacianx) [inline descope/management/access_key.py:135]: I would make a change here. Either we give the argument a default value `description: str = ""` or change the text so the word "optional" is not there since the argument will be required.
In order to keep the API without compulsory changes, I'd give a default value

File: tests/management/test_access_key.py [hunk 2]
```diff
@@ -185,6 +187,7 @@ def test_update(self):
                 client.mgmt.access_key.update,
                 "key-id",
                 "new-name",
+                "new description",
```
Comment (nmacianx) [inline tests/management/test_access_key.py:190]: If you make `description` have a default value for update, please also test the empty description is sent when no value is set for it

**Review by nmacianx — DISMISSED** [review]
LGTM! Please see minor comments

File: descope/management/access_key.py (hunk 1 above)
Comment (dorsha) [reply descope/management/access_key.py:135]: I would change to : `description(str): The description of the access key to update`, like in the id. 
So it will be consistence.

File: descope/management/access_key.py [hunk 3]
```diff
@@ -123,7 +123,7 @@ def update(
         self,
         id: str,
         name: str,
-        description: str,
+        description: Optional[str] = None,
```
Comment (nmacianx) [inline descope/management/access_key.py:126]: in this case i'd go with the default value `description: str = ""` since it will go on the payload anyway

File: descope/management/access_key.py [hunk 4]
```diff
@@ -120,6 +123,7 @@ def update(
         self,
         id: str,
         name: str,
+        description: Optional[str] = "",
```
Comment (nmacianx) [inline descope/management/access_key.py:126]: In Python, the use of `Optional[str]` means that the variable can be of type `str` or `None`. When you assign a default value of "" (an empty string), you are indicating that the variable should always be a string, even if it’s an empty one. Therefore, in this context, it would be more appropriate to use `str` rather than `Optional[str]`.

File: descope/management/access_key.py [hunk 5]
```diff
@@ -120,21 +123,22 @@ def update(
         self,
         id: str,
         name: str,
+        description: Optional[str] = None,
     ):
         """
-        Update an existing access key with the given various fields. IMPORTANT: All parameters are used as overrides
-        to the existing access key. Empty fields will override populated fields. Use carefully.
+        Update an existing access key with the given various fields. IMPORTANT: id and name are mandatory fields.
```
Comment (nmacianx) [inline descope/management/access_key.py:129]: i'd add in the comment that if description is not provided, it will not be overriden

**Review by nmacianx — DISMISSED** [review]
LGTM
